### ISSUE TITLE
bare soil handling in vic_init

### ISSUE THREAD

**Issue by jhamman** [UW-Hydro/VIC#278]
There are some a few problems with [this](https://github.com/UW-Hydro/VIC/blob/develop/vic/drivers/image/src/vic_init.c#L1050-L1067) section of code in `vic_start`.  

```C
        // Bare soil is now read in as the "last" (highest index) vegetation
        // class
        // rescale vegetation classes to 1.0 if their sum is greater than 0.99
        // otherwise throw an error
        // TBD: Need better check for equal to 1.
        if (veg_con[i][0].Cv_sum != 1.) {
            sprint_location(locstr, &(local_domain.locations[i]));
            log_warn("Cv !=  1.0 (%f) at grid cell %zd. Rescaling ...\n%s",
                     veg_con[i][0].Cv_sum, i, locstr);
            for (j = 0; j < options.NVEGTYPES; j++) {
                vidx = veg_con_map[i].vidx[j];
                if (vidx != -1) {
                    veg_con[i][vidx].Cv /= veg_con[i][0].Cv_sum;
                }
            }
            veg_con[i][0].Cv_sum = 1.;
        }
    }
```
The issues that I see are:

1.  Cv_sum does not include the fraction of the grid cell that is bare soil, so if any bare soil exists, we try to rescale, even though the comment above indicates that we should raise an error.
2.  Furthermore, if the grid cell is all bare soil (which happens frequently in the RASM domain, we end up rescaling but `Cv_sum` equals zero, leading to a divide by zero error.  

It seems to me we should follow the comment and rework this rescaling section.

**Comment by bartnijssen** [comment]
Yes - that is probably correct, although you should check whether that is true in image mode. We should carefully check the handling of the bare soil portion. The logic for this is rather confusing (this was in VIC from the beginning and way back this was partly to preserve memory). 

It would be much easier if we treated bare soil just like any other type, but that may require more changes then we want to perform at this time. In image mode we are storing the bare soil fraction in the model parameter file and can read it directly (I though it was accounted for in `Cv_sum`, but apparently not, probably to be consistent with the use of `Cv_sum` in `vic_run()`).

**Comment by tbohn** [comment]
Regarding this issue: now that I'm working with tonic, which as you might know adds an explicit bare soil class (my recent updates make this conditional on the veg library not already containing a bare soil class).  We could make an explicit bare soil class a requirement in classic too, and refer users to tonic to convert their ascii parameter files (have tonic convert to netcdf and add bare soil, then convert back to ascii).  Just a thought.

**Comment by bartnijssen** [comment]
See discussion in issue #385 .

**Comment by jhamman** [comment]
this can be closed, fixed in 3354164e34e093d813d2dadc95f69fdebfae55fe via #427
